<?xml version="1.0"?>
<launch>
    <arg name="quad_name" default="kingfisher"/>
    <arg name="use_joystick" default="False"/>
    <param name="/use_sim_time" value="True"/>
    <arg name="real_time_factor" value="1.0"/>
    
    
    
    
    
    
    
    <!-- Start Gazebo as in simulation.launch -->
    <!-- <arg name="world_name" default="$(find agiros)/resources/gazebo_files/basic.world"/>  -->
    <!-- <arg name="world_name" default="$(find agiros)/resources/gazebo_files/basic_adjusted.world"/> -->
    <!-- <arg name="world_name" default="$(find agiros)/resources/gazebo_files/basic_enclosed.world"/> -->
    <arg name="world_name" default="$(find agiros)/resources/gazebo_files/basic_enclosed_small.world"/>


    <include file="$(find agiros)/launch/simulation/base_quad_simulation.launch">
        <arg name="quad_name" value="$(arg quad_name)"/>
        <arg name="world_name" value="$(arg world_name)"/>
        <arg name="paused" value="false"/>
        <arg name="gui" value="false"/>
        <arg name="use_ground_truth" value="false"/>
        <arg name="mav_name" value="$(arg quad_name)"/>
        <arg name="model" value="$(find agiros)/resources/gazebo_files/kingfisher_model.gazebo"/>
        <arg name="enable_logging" value="false"/>
        <arg name="enable_ground_truth" value="false"/> <!-- ?? -->
        <arg name="log_file" value="$(arg quad_name)"/>
        <arg name="verbose" default="false"/>
        <arg name="debug" default="false"/>
        <arg name="x_init" value="0"/>
        <arg name="y_init" value="0"/>
    </include>



    <!-- Publish static transforms -->
    <node pkg="agiros" type="uav_tf_broadcast" name="base_link_to_world" output="screen">
        <param name="odom_topic" value="/kingfisher/agiros_pilot/state"/>
    </node>
    <!-- WRONG PREV -->
    <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="base_link_to_world"
          args="0 0 0 0 0 0 world $(arg quad_name)/base_link" /> -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="camera_to_base"
          args="0 0 0 0 0 0 $(arg quad_name)/base_link $(arg quad_name)/camera_depth_link" />     
    <node pkg="tf2_ros" type="static_transform_publisher" name="camera_optical_center_tf"
          args="0 0 0 -1.5708 0 -1.5708 $(arg quad_name)/camera_depth_link camera_depth_optical_center_link" />
    

    

    <group ns="$(arg quad_name)">
        <node name="agiros_pilot" pkg="agiros" type="agisim_node" output="screen">
            <param name="agi_param_dir" value="$(find agilib)/params"/>
            <param name="ros_param_dir" value="$(find agiros)/parameters"/>
            <param name="simulator_config" value="simulation.yaml"/>
            <param name="pilot_config" value="simple_sim_pilot.yaml"/>
            <param name="real_time_factor" value="$(arg real_time_factor)"/>
        </node>
    </group>


    <!-- Node for odom publishing used in ego-planner -->
    <node pkg="agiros" type="odom_publisher_for_ego" name="odom_publisher" output="screen"/>



    <!-- Visualization -->
    <node pkg="rviz" type="rviz" name="viz_face" args="-d $(find agiros)/resources/rviz/simulation.rviz"
          ns="$(arg quad_name)"/>

    <!-- GUI -->
    <node name="agiros_gui" pkg="rqt_gui" type="rqt_gui"
          args="-s agiros_gui.basic_flight.BasicFlight --args --quad_name $(arg quad_name)" output="screen"/>

    <!-- Gamepad and RC Utility -->
    <group if="$(arg use_joystick)">
        <include file="$(find manual_flight_interface)/launch/manual_flight_interface.launch">
            <arg name="quad_name" value="$(arg quad_name)"/>
        </include>
    </group>
</launch>
